cmd/metal-api/internal/metal/metal.go

```diff
@@ -97,6 +97,8 @@ type Entity interface {
 	GetCreated() time.Time
 	// SetCreated sets the entity's creation time
 	SetCreated(created time.Time)
+
+	TableName() string
```

Comment:
Are you sure you want the `TableName` as a method in the entity interface? You are mixing regular values of the entity (id, created, ...) with a value which is only needed for the concrete storage.

Reply:
good point, removed